tools/vim/.ycm_extra_conf.py

```diff
@@ -0,0 +1,106 @@
+# A modified example from YCM main repository
+
+import os
+import ycm_core
+
+flags = [
+'-Wall',
+'-Wextra',
+'-Werror',
+'-std=c++11',
+'-x',
+'c++'
+]
```

Comment:
Why do you need extra flags here? Shouldn't the flags be set correctly by cmake when exporting the compile command? If not, it would be nice to extract the flags that have been set by cmake, especially the std flag.